src/CMakeLists.txt

```diff
@@ -462,6 +466,10 @@ ADD_SUBDIRECTORY ( Commands )
 ADD_SUBDIRECTORY ( Desktop )
 ADD_SUBDIRECTORY ( CommandLine )
 ADD_SUBDIRECTORY ( Tests )
+if (CMAKE_COMPILER_IS_GNUCC)
```

Comment:
Looks like gcc 4.9 improves that somewhat...for c++ (by putting attributes on functions).  Unfortunately, ubuntu 14.04 LTS appears to provide gcc 4.8, so it will be a while before we can use it.